src/messages.rs

```diff
@@ -349,6 +352,50 @@ impl ToField for OutgoingMessages {
     }
 }
 
+fn encode_packet(message: &str) -> String {
+    let data = message.as_bytes();
+
+    let mut packet: Vec<u8> = Vec::with_capacity(data.len() + 4);
+
+    packet.write_u32::<BigEndian>(data.len() as u32).unwrap();
+    packet.write_all(data).unwrap();
+
+    std::str::from_utf8(&packet).unwrap().into()
+}
+
+#[derive(Default, Debug, Clone)]
+pub(crate) struct RequestHandshake {
+    min_version: i32,
+    max_version: i32,
+}
+impl RequestHandshake {
+    pub fn new(min_version: i32, max_version: i32) -> Self {
+        Self { min_version, max_version }
+    }
+    pub fn encode(&self) -> String {
+        format!("v{}..{}", self.min_version, self.max_version)
+    }
+
+    pub fn packet(&self) -> String {
```

Comment:
I've moved the length encoding from the Connection to the Message as there is `encode()` there already, and the method to produce the final bytes relies on the type of Message.